messaginginapp/src/main/java/io/customer/messaginginapp/di/DIGraphMessagingInApp.kt

```diff
@@ -1,24 +1,30 @@
 package io.customer.messaginginapp.di
 
+import io.customer.messaginginapp.MessagingInAppModuleConfig
 import io.customer.messaginginapp.ModuleMessagingInApp
-import io.customer.messaginginapp.provider.GistApi
-import io.customer.messaginginapp.provider.GistApiProvider
-import io.customer.messaginginapp.provider.GistInAppMessagesProvider
-import io.customer.messaginginapp.provider.InAppMessagesProvider
+import io.customer.messaginginapp.gist.data.listeners.GistQueue
+import io.customer.messaginginapp.gist.data.listeners.Queue
+import io.customer.messaginginapp.gist.presentation.GistProvider
+import io.customer.messaginginapp.gist.presentation.GistSdk
 import io.customer.messaginginapp.state.InAppMessagingManager
-import io.customer.sdk.CustomerIOInstance
 import io.customer.sdk.core.di.SDKComponent
 
-internal val SDKComponent.gistApiProvider: GistApi
-    get() = newInstance<GistApi> { GistApiProvider() }
+internal val SDKComponent.gistQueue: GistQueue
+    get() = singleton<GistQueue> { Queue() }
 
-internal val SDKComponent.gistProvider: InAppMessagesProvider
-    get() = newInstance<InAppMessagesProvider> { GistInAppMessagesProvider(gistApiProvider) }
+internal val SDKComponent.inAppMessaging: ModuleMessagingInApp
+    get() = ModuleMessagingInApp.instance()
 
-@Suppress("UnusedReceiverParameter")
-fun CustomerIOInstance.inAppMessaging(): ModuleMessagingInApp {
```

Comment:
Why did we remove this? This will break public facing API again for dismissing in-app message 🤐

Reply:
good point